render/src/lib/views/sidebar.vertical/port.available/components.ts

```diff
@@ -0,0 +1,344 @@
+// tslint:disable:no-inferrable-types
+
+import { Component, OnDestroy, Input, AfterViewInit, ViewChild, ElementRef, ChangeDetectorRef, OnInit } from '@angular/core';
+import { IPortInfo } from '../../../common/interface.portinfo';
+import Chart from 'chart.js';
+import { Subscription, Subject } from 'rxjs';
+import Service, { IPort } from '../../../services/service';
+import { EHostEvents } from '../../../common/host.events';
+import { SidebarVerticalPortDialogComponent } from '../port.options/component';
+import { IOptions, CDefaultOptions } from '../../../common/interface.options';
+import { ENotificationType } from 'chipmunk.client.toolkit';
+import { SidebarVerticalPortWarningComponent } from '../port.warning/component';
+
+const LIMIT = 500;
+
+@Component({
+    selector: 'lib-dia-port-available-com',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class DialogAvailablePortComponent implements OnDestroy, AfterViewInit, OnInit {
+
+    @Input() port: IPortInfo;
+    @Input() observables: {tick: Subject<boolean>, resize: Subject<{ sidebar_width: number, sidebar_height: number }>};
+    @ViewChild('canvas') canvas: ElementRef<HTMLCanvasElement>;
+
+    private _subscriptions: { [key: string]: Subscription } = {};
+    private _ctx: any;
```

Comment:
I think you don't need to store context `_ctx`, you need it just once during creating of chart. But no needs to store it